patches/minecraft/net/minecraft/world/World.java.patch

```diff
@@ -17,28 +17,29 @@
      private int field_181546_a = 63;
      protected boolean field_72999_e;
      public final List<Entity> field_72996_f = Lists.<Entity>newArrayList();
-@@ -108,6 +115,12 @@
+@@ -108,6 +115,13 @@
      private final WorldBorder field_175728_M;
      int[] field_72994_J;
  
 +    public boolean restoringBlockSnapshots = false;
 +    public boolean captureBlockSnapshots = false;
 +    public java.util.ArrayList<net.minecraftforge.common.util.BlockSnapshot> capturedBlockSnapshots = new java.util.ArrayList<net.minecraftforge.common.util.BlockSnapshot>();
++    private java.util.Set<net.minecraft.util.math.ChunkPos> tileEntitiesChunkToBeRemoved = new java.util.HashSet<>();
```

Comment:
Couldn't a `LongOpenHashSet` (from fastutils) or `TLongHashSet` (from trove) be used here? Note that while the `LongOpenHashSet` is probably faster, it's not available in 1.11...
To get the `long`, there's `ChunkPos.asLong(int, int)`